concert/experiments/imaging.py

```diff
@@ -60,3 +63,555 @@ def tomo_max_speed(frame_width, frame_rate):
+    """
+    def __init__(self, shutter):
+        self._shutter = shutter
+
+    async def start_sample_exposure(self):
+        if await self._shutter.get_state() != "open":
+            await self._shutter.open()
+
+    async def stop_sample_exposure(self):
+        if await self._shutter.get_state() != "closed":
+            await self._shutter.close()
+
+
+class XrayTubeMixin:
+    """
+    Mixin to implement the required function start_sample_exposure and stop_sample_exposure in the imaging experiments
+    by switching the X-ray tube on and off.
+    """
+    def __init__(self, xray_tube):
+        self._xray_tube = xray_tube
+
+    async def start_sample_exposure(self):
+        if await self._xray_tube.get_state() != "on":
+            await self._xray_tube.start()
+
+    async def stop_sample_exposure(self):
+        if await self._xray_tube.get_state() == "on":
+            await self._xray_tube.stop()
+
+
+class Radiography(Experiment):
+    """
+    Radiography experiment
+
+    This records dark images (without beam) and flat images (with beam and without the sample) as well as the
+    projections with the sample in the beam.
+    """
+    num_flats = Parameter()
+    num_darks = Parameter()
+    num_projections = Parameter()
+    radio_position = Quantity(q.mm)
+    flat_position = Quantity(q.mm)
+
+    def __init__(self, walker, flat_motor, radio_position, flat_position, camera,
+                 num_flats=200, num_darks=200, num_projections=3000, separate_scans=True):
+        self._num_flats = num_flats
+        self._num_darks = num_darks
+        self._num_projections = num_projections
+        self._radio_position = radio_position
+        self._flat_position = flat_position
+        self._finished = None
+        self._flat_motor = flat_motor
+        self._camera = camera
+        darks_acq = Acquisition("darks", self._take_darks)
+        flats_acq = Acquisition("flats", self._take_flats)
+        radios_acq = Acquisition("radios", self._take_radios)
+        super(Radiography, self).__init__([darks_acq, flats_acq, radios_acq], walker, separate_scans=separate_scans)
+
+    async def _get_num_flats(self):
+        return self._num_flats
+
+    async def _get_num_darks(self):
+        return self._num_darks
+
+    async def _get_num_projections(self):
+        return self._num_projections
+
+    async def _get_radio_position(self):
+        return self._radio_position
+
+    async def _get_flat_position(self):
+        return self._flat_position
+
+    async def _set_num_flats(self, n):
+        self._num_flats = int(n)
+
+    async def _set_num_darks(self, n):
+        self._num_darks = int(n)
+
+    async def _set_num_projections(self, n):
+        self._num_projections = int(n)
+
+    async def _set_flat_position(self, position):
+        self._flat_position = position
+
+    async def _set_radio_position(self, position):
+        self._radio_position = position
+
+    async def _prepare_flats(self):
```

Comment:
Is there a reason why these should be private?

Reply:
The property setter is always private. 

I thought the prepare/finish- functions would be only called from within the Experiment class. 
If you they should be called by the User, I can make them public.

Reply:
I think the comment is on `_prepare_flats`.

Reply:
Do we want to have them public?